src/functions_framework/__init__.py

```diff
@@ -59,51 +70,100 @@ def __init__(
             }
         self.data = data
 
-
 def _http_view_func_wrapper(function, request):
     def view_func(path):
         return function(request._get_current_object())
 
     return view_func
 
 
-def _is_binary_cloud_event(request):
-    return (
-        request.headers.get("ce-type")
-        and request.headers.get("ce-specversion")
-        and request.headers.get("ce-source")
-        and request.headers.get("ce-id")
-    )
+def _get_cloud_event_version():
+    return v1.Event()
```

Comment:
```diff
- def _get_cloud_event_version():
-     return v1.Event()
+ def _get_cloud_event_version(request):
+     ...
```
We should be able to infer the event version from the request. If it's not a V1 event, I'm not sure what should happen (maybe `abort(400)`?), but this would be the right place to check it.